tests/func/remote/test_setting_paths.py

```diff
@@ -0,0 +1,37 @@
+import os
+import posixpath
+
+import configobj
+import pytest
+
+from dvc.main import main
+from dvc.repo import Repo
+
+
+@pytest.mark.parametrize(
+    "field, remote_url",
+    [
+        ("credentialpath", "s3://mybucket/my/path"),
+        ("credentialpath", "gs://my-bucket/path"),
+        ("keyfile", "ssh://user@example.com:1234/path/to/dir"),
+        ("cert_path", "webdavs://example.com/files/USERNAME/"),
+        ("key_path", "webdavs://example.com/files/USERNAME/"),
+    ],
+)
+def test_relative_config_setting(tmp_dir, dvc, field, remote_url):
+    a_file = os.path.join("secrets", "file.txt")
+
+    remote_name = "some_name"
+    assert main(["remote", "add", "-d", remote_name, remote_url]) == 0
+    assert main(["remote", "modify", remote_name, field, a_file]) == 0
+
+    # We need to load repo again to test updates to the config
+    str_path = os.fspath(tmp_dir)
+    repo = Repo(str_path)
+
+    # Check that in config we got the path relative to the config file itself
+    # Also, we store posix path even on Windows
+    config = configobj.ConfigObj(repo.config.files["repo"])
+    assert config[f'remote "{remote_name}"'][field] == posixpath.join(
+        "..", "secrets", "file.txt"
+    )
```

Comment:
Seems like this is testing how it is written to the config file, but not really how it is loaded. Or am I missing something?